[AY1920S1-CS2113-T14-2] restaurant manager

9hafidz6 opened the pull request.

sanjukta99 commented on src/main/java/duke/Dishes/DishList.java:
```diff
@@ -0,0 +1,36 @@
+import java.util.List;
+
+public class DishList {
+    private List<Dishes> dishlist;
```
Important that you write proper JavaDoc comments for all classes. You can find the Java Coding Standards on the module website

sanjukta99 commented on src/main/java/duke/Dishes/Dishes.java:
```diff
@@ -0,0 +1,53 @@
+
+//    public String toString() {
+//        return dishname;
+//    }
```
It's best to delete commented out code

sanjukta99 commented on src/main/java/duke/command/Cmd.java:
```diff
@@ -0,0 +1,18 @@
+import duke.ui.Ui;
+
+public abstract class Cmd<T> {
+    public abstract void execute(T tasklist, Ui ui, Storage storage) throws DukeException;
```
Maybe using a more descriptive name like `Command` might be better?

sanjukta99 commented on src/main/java/duke/command/dishesCommand/AddDishCommand.java:
```diff
@@ -0,0 +1,54 @@
+    private Dishes dish;
+    private int amount;
+    private int Nb;
+
```
What is `Nb`? Better to use a more descriptive name. That way even someone who's reading your code for the first time can understand what's going on in that part of the code

sanjukta99 commented on src/main/java/duke/parser/Convert.java:
```diff
@@ -0,0 +1,78 @@
+ * Convert between String and Date.
+ */
+public class Convert {
+
```
Convert what? Consider renaming to something more descriptive

sanjukta99 commented on src/test/FridgeTest.java:
```diff
@@ -0,0 +1,69 @@
+    Fridge fridge = new Fridge();
+    @Test
+    void testPutIngredient() throws DukeException {
+        fridge = new Fridge();
```
Names for tests should follow the format **featureUnderTest_testScenario_expectedBehavior()**. Update accordingly for all the testcases

sanjukta99 reviewed (commented):
Overall, good progress. Take a look at my suggestions. 

You should follow the Java Coding Standards for JavaDoc comments and naming conventions. We also recommend that you delete commented out code as it makes the code more readable.

9hafidz6 committed:
updated UG, modified parser  (#59)

* updated UG
implemented while loop in Duke class for Dish
modified parser to handle more errors

CEGLincoln committed:
did i not

saradj committed:
Update UserGuide.md

saradj committed:
Update UserGuide.md

saradj committed:
Update DeveloperGuide.md

Commit:
update add order, done order, alter order commands

saradj committed:
Update AddCommand.java

Commit:
update list order and remove order command

x3chillax committed:
updated DG and UG

x3chillax committed:
Merge branch 'master' into master

x3chillax committed:
Merge pull request #60 from x3chillax/master

updated DG and UG

Commit:
fix compile error and update delete order, list order ... commands

JiahanYu committed:
Merge pull request #2 from AY1920S1-CS2113-T14-2/master

update multi-level command

Commit:
update Ui class

Commit:
fix compile problem regarding to alter order date.
update Parser, Ui, and Duke main class.

Commit:
optimize imports

JiahanYu committed:
Merge pull request #61 from VirginiaYu/master

update order feature

CEGLincoln committed:
nothing

CEGLincoln committed:
So many conflictsgit add .

CEGLincoln committed:
Merge pull request #78 from CEGLincoln/master

Updating my end

JiahanYu committed:
Merge pull request #3 from AY1920S1-CS2113-T14-2/master

to fix issues reported by PED run

Commit:
InitOrderList Command

Commit:
1. Solve order template duplicate issues.
2. Format order command to make sense and better user experience.
3. Add functions to check for valid date/month/year.
4. Fix update problem in setting an order date.
5. Extract Parser.parse function case "c" to several subfunctions--look neat.
6. Fix crashes when converting a string to date.
7. Fix small typo when printing out messages.

JiahanYu committed:
Merge pull request #104 from VirginiaYu/master

More neat order command and handle exceptions

Commit:
type

Commit:
solved -- Entering an incorrect response to "Are you sure you want to clear list" provides no visual indication #87

Commit:
put fridge.txt into source folder

Commit:
optimize imports

Commit:
add javaDoc to initOrderListCommand class

Commit:
make print in file format in one line for an order.
update & add order info to order.txt √

Commit:
order storage loading from txt file
order storage update to txt file

Commit:
rename Cmd to Command, since the tutor says it make sense

Commit:
todo list not finished yet

Commit:
add todo list feature

Commit:
finish todo list, add one more option to view the latest task list

JiahanYu committed:
Merge pull request #108 from VirginiaYu/master

fix order read/write, solve some issues reported by PED run

CEGLincoln committed:
Huge Changes

CEGLincoln committed:
whats the issue

CEGLincoln committed:
commit again

CEGLincoln committed:
it compiles!

CEGLincoln committed:
hope I didnt break anything

saradj committed:
Merge pull request #110 from CEGLincoln/master

Changed the command system

CEGLincoln committed:
Solved your issues

CEGLincoln committed:
Merge remote-tracking branch 'upstream/master'
No its not

saradj committed:
Merge pull request #1 from AY1920S1-CS2113-T14-2/master

getting the update

CEGLincoln committed:
Merge pull request #111 from CEGLincoln/master

Solved your issues

saradj committed:
fixing removeAllExpired and UseIngredient

saradj committed:
Merge branch 'master' into master

CEGLincoln committed:
Merge pull request #112 from saradj/master

Changed the argument of execute for Command to Fridge, and fixed RemoveAllExpired and UseCommand

CEGLincoln committed:
improve listdishcommand

CEGLincoln committed:
Merge pull request #113 from CEGLincoln/master

improve listdishcommand

CEGLincoln committed:
fix bug

CEGLincoln committed:
Merge pull request #114 from CEGLincoln/master

fix bug

Commit:
Update UG, cleaning format. Hope every teammate can take a look.

Commit:
add ui.showlines to format

Commit:
udpate UG order command and todo list.

Commit:
no message

JiahanYu committed:
Merge branch 'master' into master

JiahanYu committed:
Merge pull request #116 from VirginiaYu/master

Update UG

x3chillax committed:
no message

x3chillax committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-2/main1

# Conflicts:
#	data/fridge.txt
#	src/main/java/duke/command/ingredientCommand/FindToday.java

CEGLincoln committed:
fix loading arror from hard disk

CEGLincoln committed:
Merge pull request #117 from CEGLincoln/master

fix loading arror from hard disk

x3chillax committed:
added commands to change ingredient name/amount using index number.

x3chillax committed:
Merge pull request #118 from x3chillax/master

added commands to change ingredient name/amount using index number.

x3chillax committed:
added things to template in UI

x3chillax committed:
Merge pull request #119 from x3chillax/master

added things to template in UI

9hafidz6 committed:
added 2 new features to dishcommand (#120)

* UG update and new UI

* update UG

* added javadoc comments on some methods

* updated some commands

* added more javadoc comments
tried to improve recipeStorage

* added Junit test for dishcommands

* touch up

* added 2 features to dishcommand, find and changename
update UG and DG

Commit:
* Fix compile error, fix bugs
* update todo list in accordance with order update
* Enhance read/write operations

9hafidz6 committed:
changed ui

x3chillax committed:
code cleanup. removed those tasks thingy

x3chillax committed:
Merge pull request #121 from x3chillax/master

code cleanup. removed those tasks thingy

Commit:
fix todolist error

JiahanYu committed:
Merge branch 'master' into master

JiahanYu committed:
Merge pull request #122 from VirginiaYu/master

Fix several errors, such as NullException. Update Todolist accordingly with the order management.

JiahanYu committed:
fix error

JiahanYu committed:
fix read/write error

JiahanYu committed:
More explicit reminder

JiahanYu committed:
showline after exception examination

JiahanYu committed:
.

JiahanYu committed:
Update UG

fix fault sample output and non-correlation btw doc and code

JiahanYu committed:
.

JiahanYu committed:
Fix indentation

JiahanYu committed:
.

JiahanYu committed:
Merge pull request #123 from VirginiaYu/master

update UG.md and fix small bugs

CEGLincoln committed:
what is this

CEGLincoln committed:
fix m conflict

JiahanYu committed:
.

JiahanYu committed:
Update order template

JiahanYu committed:
changing to "✓" : "✗"

JiahanYu committed:
compile error

saradj committed:
Merge pull request #2 from AY1920S1-CS2113-T14-2/master

getting the update

9hafidz6 committed:
updated UG,DG. able to read from recipebook.txt (#125)

* modified some of the dish commands to handle duplicates (dish class)
added Javadoc comments
updated UG, DG
attempt to use recipe storage

* Now able to read from recipebook.txt

saradj committed:
Adding JavaDoc and minor fix

saradj committed:
Merge branch 'master' into master

saradj committed:
Merge pull request #126 from saradj/master

Adding JavaDoc and fixing some minor issues

saradj committed:
Merge pull request #3 from AY1920S1-CS2113-T14-2/master

Merge pull request #126 from saradj/master

saradj committed:
fixing merge problems

saradj committed:
Merge pull request #127 from saradj/master

Fixing merge problems...

saradj committed:
Update DukeException.java

saradj committed:
update ui to match user guide

CEGLincoln committed:
why still have merge conflict

CEGLincoln committed:
minor improvements

CEGLincoln committed:
Merge pull request #128 from CEGLincoln/master

Minor Improvements

9hafidz6 committed:
updated UG,DG. dishesCommand able write/delete/update to file (#129)

* Now able to read from recipebook.txt

* updated dish commands to write/change to recipebook.txt
added some UI elements for each template

* updated UG and DG, content page updated, manual testing etc
now the dishesComands are able to write/update to hard disk

saradj committed:
Change UG and minor change to io flow

saradj committed:
Merge branch 'master' into master

saradj committed:
Merge pull request #130 from saradj/master

Change UG and minor change to io flow for the Fridge interaction

saradj committed:
Merge pull request #4 from AY1920S1-CS2113-T14-2/master

Merge pull request #130 from saradj/master

9hafidz6 committed:
added back previous table of content

9hafidz6 committed:
added back previous table of content for UG

saradj committed:
DG update

saradj committed:
Merge pull request #131 from saradj/master

DG update did some changes in component classification!

x3chillax committed:
added documentation

saradj committed:
UG DG update

x3chillax committed:
added documentation

x3chillax committed:
Merge pull request #132 from x3chillax/master

added documentation

x3chillax committed:
some updates to DG

x3chillax committed:
Merge pull request #133 from x3chillax/master

some updates to DG

saradj committed:
Cleaning code

saradj committed:
Merge branch 'master' into master

saradj committed:
Merge pull request #134 from saradj/master

Completing Javadoc, UG and DG update

saradj committed:
Merge pull request #5 from AY1920S1-CS2113-T14-2/master

Merge pull request #134 from saradj/master

saradj committed:
Update IngredientsList.java

JiahanYu committed:
Merge branch 'master' into master

JiahanYu committed:
Merge pull request #4 from AY1920S1-CS2113-T14-2/master

Pull Request from group repo

saradj committed:
DG update

saradj committed:
updating ug

saradj committed:
Merge pull request #135 from saradj/master

Update DG

saradj committed:
Merge pull request #6 from AY1920S1-CS2113-T14-2/master

update

Commit:
Update author and delete some garbage code

saradj committed:
update

saradj committed:
Merge pull request #136 from saradj/master

Updating the DG

Commit:
Add JavaDoc for classes

saradj committed:
Update Duke.java

Commit:
Add test code for order class

x3chillax committed:
updated UG

Commit:
Add test code file for TodayTodoList

x3chillax committed:
Merge branch 'master' into master

x3chillax committed:
Merge pull request #137 from x3chillax/master

updated UG

Commit:
update/optimize two test files imports

JiahanYu committed:
Merge pull request #138 from VirginiaYu/master

Tail-in works

x3chillax committed:
updated UG

x3chillax committed:
Merge pull request #139 from x3chillax/master

updated UG

x3chillax committed:
updated DG

x3chillax committed:
Merge pull request #140 from x3chillax/master

updated DG

Commit:
Update ugly indentation when drawing those "order" "ingredient" "dish" stuffs
keep it as same indentation as showline!

Commit:
Exception Message Update, corresponds to UG and making it more organized.

Commit:
Update UG
- Add content link
- Update order component details
- Add features and ideas can come in v2.0
- Also show exception messages
- Add bug assignee

Commit:
format

JiahanYu committed:
Merge branch 'master' into master

JiahanYu committed:
Merge pull request #141 from VirginiaYu/master

Update UG, tiny modification on code

JiahanYu committed:
Update AboutUs.md

JiahanYu committed:
Not showing characters when showing template

JiahanYu committed:
Add "order" drawing when step into submenu

JiahanYu committed:
Add Storage png

JiahanYu committed:
Update DG

Update DG Storage part and add diagram

JiahanYu committed:
Update Dg

JiahanYu committed:
delete storage.png

should update

JiahanYu committed:
add storage updated figure

JiahanYu committed:
Update path to png

saradj committed:
update

JiahanYu committed:
Upload storage.png

JiahanYu committed:
Merge pull request #143 from VirginiaYu/master

Update DG

saradj committed:
Merge branch 'master' into master

JiahanYu committed:
Update DG for order ui

saradj committed:
Merge branch 'master' into master

saradj committed:
Merge pull request #142 from saradj/master

some ui changes for the fridge

saradj committed:
ug update

JiahanYu committed:
Upload Command.png

saradj committed:
Merge branch 'master' of https://github.com/saradj/main

JiahanYu committed:
Update DG

Update Command Component and add graphs of the component

saradj committed:
adding assertions

saradj committed:
Merge pull request #144 from saradj/master

Asserts and preventing use 0 of an ingredient!

JiahanYu committed:
Upload ListComponent.png

JiahanYu committed:
Update UG.md

Correct "<>" in md to "\<>" so as to recognize.
Update List component and make the methods the newest version.

JiahanYu committed:
old version png removed

JiahanYu committed:
Add new version of list component figure

JiahanYu committed:
Merge pull request #145 from VirginiaYu/master

Update UG.md

x3chillax committed:
no message

x3chillax committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-2/main1

# Conflicts:
#	fridge.txt

JiahanYu committed:
Command figure update

JiahanYu committed:
delete unnecessary list command

JiahanYu committed:
update its inheritance to command class

Commit:
BIG CHANGE: put all lists into list packages, similar to `Storage` component structure.

JiahanYu committed:
delete command component graph

because the list command is deleted

JiahanYu committed:
upload png

manually solve conflict

Commit:
update with newest version

Commit:
to fix merge conflict (roll back)

JiahanYu committed:
Merge pull request #146 from VirginiaYu/master

BIG CHANGE on list packages

Commit:
move ingredient list into list packages

JiahanYu committed:
Merge pull request #147 from VirginiaYu/master

fix merge confilct & update list package

x3chillax committed:
DG update

x3chillax committed:
Merge pull request #148 from x3chillax/master

update DG

saradj committed:
Merge pull request #7 from AY1920S1-CS2113-T14-2/master

update

saradj committed:
fixing compile err

saradj committed:
Merge pull request #149 from saradj/master

Fixing imports

x3chillax committed:
no message

saradj committed:
update dg

saradj committed:
Merge pull request #150 from saradj/master

update dg

saradj committed:
Add files via upload

saradj committed:
Update DeveloperGuide.md

saradj committed:
Add files via upload

saradj committed:
Delete PrintableUML.png

saradj committed:
Add files via upload

saradj committed:
Add files via upload

saradj committed:
Update DeveloperGuide.md

saradj committed:
Add files via upload

saradj committed:
Update DeveloperGuide.md

saradj committed:
Add files via upload

saradj committed:
Delete StorageUML.png

saradj committed:
Add files via upload

saradj committed:
dg

saradj committed:
Merge pull request #151 from saradj/master

adding UML

saradj committed:
Update DeveloperGuide.md

saradj committed:
uml:

saradj committed:
Merge branch 'master' into master

saradj committed:
Merge pull request #152 from saradj/master

uml:

9hafidz6 committed:
updated UG and DG. added some Junit test for Dish (#153)

* updated images so that when converted to pdf image is retained (UG and DG)

* added more jUnit test case for  dish

* updated UG DG

JiahanYu committed:
Merge pull request #5 from AY1920S1-CS2113-T14-2/master

update

9hafidz6 committed:
added parser image to DG

9hafidz6 committed:
updated UG and additional javadoc  (#154)

* additional Javadoc

* updated UG

9hafidz6 committed:
minor update to DG and UG (#155)

* minor update to UG and DG

9hafidz6 committed:
fixed UI image on UG

saradj committed:
Merge pull request #8 from AY1920S1-CS2113-T14-2/master

update

saradj committed:
update ug dg

saradj committed:
ug dg update

9hafidz6 committed:
new Ui.png, t command

saradj committed:
added a check for ingName length

x3chillax committed:
minor changes to DG

x3chillax committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-2/main1

x3chillax committed:
Merge pull request #157 from x3chillax/master

some changes to DG

saradj committed:
dg update

9hafidz6 committed:
changed quick start to java -jar v1.4

saradj committed:
Merge pull request #156 from saradj/master

added a check for ingredient name length,

9hafidz6 committed:
updated DG, the output of recipebook.txt

9hafidz6 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-2/main

x3chillax committed:
DG update

9hafidz6 committed:
merged with sara checkIngNameLength()

x3chillax committed:
Merge pull request #158 from x3chillax/master

DG update

9hafidz6 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T14-2/main

JiahanYu committed:
Link to profolio

will upload ppp later

JiahanYu committed:
add paranthesis

saradj committed:
Add files via upload

saradj committed:
Update DeveloperGuide.md

saradj committed:
Update DeveloperGuide.md

JiahanYu committed:
Update DG

JiahanYu committed:
Merge branch 'master' into master

JiahanYu committed:
Merge pull request #159 from VirginiaYu/master

Update dg

saradj committed:
Update DeveloperGuide.md
